### PR TITLE
Fedn compute package

### PR DESCRIPTION
**Pull request by morganekmefjord**
This new structure supports a remote distributable compute_package structure which will be set on reducer startup and subsequent client connected will receive and use the compute package distributed.

Client default is remote distributed.

## Solution description
Compute bundles can now be seeded with the alliance and are remotely distributed to connecting clients.
This allows for a much leaner approach on client side ensuring that only dataset is required to get started.

## Solution enhanced

**CLI additions**
- cli now has additional commands to build a compute package.
cli requires a default structure. Please see test/mnist/package/ folder for a package example structure.

**Reducer additions**
- settings.yaml
Reducer requires a settings.yaml for seeding the initial values for reducer.

**Combiner**
No changes applicable.

**Client**
Clients spawn now with remote to true meaning that you only are required to point to the dataset and the rest of the parameters and setup is taken care of by FEDn, both execution parameters, bundles etc.

Client now have additional parameters

## Further work
- Ensure backend data instantiation / example: "s3 initial variables" are also derived from settings.yaml
- Client dry run to check requirements are added to args but not yet implemented.
- Combiner configurations and customized strategies _could_ be sent in the same manner as well.

### CODE REVIEW

File: test/mnist/package/client/fedn.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+entry_points:
+  predict:
+    command: python3 predict.py
+  train:
+    command: python3 train.py
+  validate:
+    command: python3 validate.py
```
Comment (ahellander) [inline test/mnist/package/client/fedn.yaml:7]: Now we have duplication of: 

test/mnist/package/client/train.py etc
and 
test/mnist/client/train.py etc. 

There are updates in the latter in develop that needs to be ported to the new package folder structure. Then it should be deleted?

File: fedn/fedn/client.py [hunk 2]
```diff
@@ -85,16 +85,27 @@ def __init__(self, config):
         print("Client: {} connected {} to {}:{}".format(self.name,
                                                         "SECURED" if client_config['certificate'] else "INSECURE",
                                                         client_config['host'], client_config['port']), flush=True)
+        if config['remote_compute_context']:
```
[the remark this answers is not in the document]
Comment (morganekmefjord) [reply fedn/fedn/client.py:88]: Redistribution on client restart is by design currently as we need to ensure we are always working with the latest package on client startup.

File: test/mnist/package/client/fedn.yaml (hunk 1 above)
Comment (morganekmefjord) [reply test/mnist/package/client/fedn.yaml]: Updated.
- Remove duplicate code
- Updated package method with better verbosity.

**Review by ahellander — APPROVED** [review]
I think we should delete the 

/test/mnist/package/client 

folder, to avoid confusion. The package is based on 
/test/mnist/client

as it should be.

File: client.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      context: .
+      dockerfile: components/client/Dockerfile
+    working_dir: /app/client
+    command: /bin/bash -c "tail -f /dev/null"
+    volumes:
+      #- ./test/${EXAMPLE}/client:/app/client
+      #- ./test/${EXAMPLE}/data:/app/data
+      - ./test/${EXAMPLE}/reducer/certs/reducer-cert.pem:/app/client/certs/client-cert.pem
+    ports:
+      - 8080:8080
```
[the remark this answers is not in the document]
Comment (morganekmefjord) [reply client.yaml:24]: Well. Clients run a WEB GUI to show status of client work. It is good to show every client has such capability with one client enabled to expose the GUI by default.

**Pull request merged**